src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -59,15 +61,33 @@ public class CISShadowMonitor {
     private static final String CIS_SHADOW_SUFFIX = "-gci";
     private static final String VERSION = "version";
     private static final long TIMEOUT_FOR_SUBSCRIBING_TO_TOPICS_SECONDS = Duration.ofMinutes(1).getSeconds();
-    private static final long WAIT_TIME_TO_SUBSCRIBE_AGAIN_IN_MS = Duration.ofMinutes(2).toMillis();
-    private static final Random JITTER = new Random();
     private static final String SHADOW_UPDATE_DELTA_TOPIC = "$aws/things/%s/shadow/update/delta";
     private static final String SHADOW_GET_ACCEPTED_TOPIC = "$aws/things/%s/shadow/get/accepted";
 
-    private static final RetryUtils.RetryConfig GET_CONNECTIVITY_RETRY_CONFIG = RetryUtils.RetryConfig.builder()
-            .initialRetryInterval(Duration.ofMinutes(1L)).maxRetryInterval(Duration.ofMinutes(30L))
-            .maxAttempt(Integer.MAX_VALUE).retryableExceptions(Arrays.asList(ThrottlingException.class,
-                    InternalServerException.class)).build();
+    @Setter(AccessLevel.PACKAGE) // for unit testing
+    private Duration subscribeInitialRetryInterval = Duration.ofMinutes(1);
+
+    private final Supplier<RetryUtils.RetryConfig> subscribeRetryConfig = () ->
+            RetryUtils.RetryConfig.builder()
+                    .initialRetryInterval(subscribeInitialRetryInterval)
+                    .maxRetryInterval(Duration.ofMinutes(2))
+                    .maxAttempt(Integer.MAX_VALUE)
```

Comment:
set a high max attempts to keep parity with original code, but i feel like we should fail more quickly.  open to thoughts on this.